client/python/test/test_client.py

```diff
@@ -157,6 +157,16 @@ def test_job_output_video(self):
         )
         assert pathlib.Path(job.result()).exists()
 
+    def test_progress_updates(self):
+        client = Client(src="gradio-tests/progress-tracker")
+        job = client.submit("hello", api_name="/predict")
+        statuses = []
+        while not job.done():
+            statuses.append(job.status())
+            time.sleep(0.05)
+        assert any(s.code == Status.PROGRESS for s in statuses)
+        assert any(s.progress_data is not None for s in statuses)
```

Comment:
Nit: can we make this test more speicific? E.g. assert something like the 
```
any(s.progress_data == ProgressUnit(index=19, length=20, unit="steps", progress=None, desc=None) for s in statuses)
```
It will help readers of the codebase understand the expected format of the progress data